code/log/lib/Loot/Log/Rustic.hs

```diff
@@ -0,0 +1,62 @@
+{-# LANGUAGE CPP #-}
+
+-- | Down to earth logging for the lazy ones
+--
+-- No configuration, just use functions from this module instead of @putStrLn@
+-- and they will do the rest. All messages above @Debug@ are printed to @stderr@.
```

Comment:
I'd rather use global variable to control this behaviour. We can present a function `setDebug :: IO ()` that enables debug output by enabling a parameter inside `MVar`. Is this approach any better?

Reply:
Well, I was thinking about it and I am not sure tbh. My view on this is that having a global variable no longer feels that dumb-simple and pure, while the only benefit it gives is a sort of dynamic configuration, e.g. with a command-line switch, and if you want this, it probably means that it’s time to switch to a proper logging implementation. This one is meant only for the simplest things, one-off scripts, etc.

Reply:
Compilation flags look scary to me -- of course, they are easy to configure, but I need to look up the exact way to do it. Instead, adding a line of `hs` code enabling debug output is straightforward. The user doesn't need to know how exactly it is implemented under the bonnet.

Reply:
I really think that for simplest things and one-off scripts it's not the compilation flag which should be used, but an environment variable. If you're quickly prototyping something, you really don't want to recompile the thing to enable some debug logs for the next run.

Reply:
CPP also looks scary to me, because I need to compile a _library_ with `-DDEBUG`, not my application, AFAIU. Even if I am wrong, even the need to think about it doesn't sound good.
+1 to environment variable.